### PR TITLE
Implement module.to_empty()

### LINKED ISSUE #54600: Support nn.Module.to(..., copy_data=False)
The motivation is that we are going to support allocating nn.Modules on meta device, and you cannot "convert" a tensor from meta to cpu (because the data is unspecified). But sometimes it's still useful to inplace transmute the parameters from meta to cpu. `copy_data=False` would let you do this transmutation, while also acknowledging that the old data doesn't get preserved (in the case of meta, this is because it doesn't exist, but in the case of other devices, we would just not copy).

cc @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by jbschlosser**
Fixes #54600

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -700,6 +700,18 @@ def bfloat16(self: T) -> T:
         """
         return self._apply(lambda t: t.bfloat16() if t.is_floating_point() else t)
 
+    def empty(self: T, device: Optional[Union[str, device]] = None) -> T:
+        r"""Moves the parameters and buffers to the specified device without copying storage.
+
+        Args:
+            device (:class:`torch.device`): The desired device of the parameters
+                and buffers in this module. If None, the current device is used. Default: None
```
Comment (albanD) [inline torch/nn/modules/module.py:708]: What does it mean to call this with `None` argument? Is it like a clone but generating random value? Is there any valid use case for that or do we want to make device a required argument?

Comment (ezyang) [reply torch/nn/modules/module.py:708]: Yeah, it's like clone but without initializing any of the new weights. Not sure if there is a valid use case.

Comment (jbschlosser) [reply torch/nn/modules/module.py:708]: Can't think of a valid use case either! The modules ends up with "random" (uninitialized) values for all params / buffers. 

I'm good with making `device` required.

File: torch/nn/modules/module.py [hunk 2]
```diff
@@ -700,6 +700,18 @@ def bfloat16(self: T) -> T:
         """
         return self._apply(lambda t: t.bfloat16() if t.is_floating_point() else t)
 
+    def empty(self: T, device: Optional[Union[str, device]]) -> T:
```
Comment (albanD) [inline torch/nn/modules/module.py:703]: nit: type shouldn't be optional anymore.

Also, maybe kwarg only?

**Review by albanD — APPROVED** [review]
LGTM !

**Pull request closed without merging**